modules/src/ics04_channel/context.rs

```diff
@@ -60,16 +60,40 @@ pub trait ChannelKeeper {
                 )?;
             }
             State::TryOpen => {
-                self.store_channel(
-                    &(result.port_id, result.channel_id.clone().unwrap()),
-                    &result.channel_end,
-                )?;
-                // TODO: If this is the first time the handler processed this channel, associate the
-                // channel end to its client identifier.
-                // self.store_channel_to_connection(
-                //     &(result.port_id,result.channel_id),
-                //     &result.... ,
-                // )?;
+                if result.channel_id.is_none() {
+                    // If this is the first time the handler processed this channel
+                    let channel_id = self.next_channel_id();
+
+                    self.store_channel(
+                        &(result.port_id.clone(), channel_id.clone()),
+                        &result.channel_end,
+                    )?;
+
+                    // associate also the channel end to its connection
+                    self.store_connection_channels(
+                        &result.channel_end.connection_hops()[0].clone(),
+                        &(result.port_id.clone(), channel_id.clone()),
+                    )?;
+
+                    // initialize send sequence number
+                    self.store_next_sequence_send(
+                        &(result.port_id.clone(), channel_id.clone()),
+                        1,
+                    )?;
+                    // initialize recv sequence number
+                    self.store_next_sequence_recv(
+                        &(result.port_id.clone(), channel_id.clone()),
+                        1,
+                    )?;
+                    // initialize ack sequence number
+                    self.store_next_sequence_ack(&(result.port_id.clone(), channel_id), 1)?;
+                } else {
+                    //the handler processed this channel for channel open init
+                    self.store_channel(
+                        &(result.port_id.clone(), result.channel_id.clone().unwrap()),
+                        &result.channel_end,
+                    )?;
+                }
```

Comment:
I think this can stay as the body of `store_channel_result()`, i.e. remove the `match`.

Reply:
The spec says that if there is no channel open init, the chan open try needs to do the initialisation of the packet sequence and adding the other stores besides the channel.

Reply:
Yes and it's the same for `open init`. I was just wondering if we could just do this with:
```
 if result.channel_id.is_none() {
...
```
So we would trust that `process` did the right thing.